tasks/configure-xml.yml

```diff
@@ -44,4 +52,4 @@
 # you cannot "notify" here because zkfc is not enabled when standalone
 - set_fact:
     hadoop_config_changed: true
-  when: "{{ core_site.changed or hdfs_site.changed or mapred_site.changed or yarn_site.changed or (hadoop_config_changed is defined and hadoop_config_changed) }}"
+  when: core_site.changed or hdfs_site.changed or mapred_site.changed or yarn_site.changed or (hadoop_config_changed is defined and hadoop_config_changed)
```

Comment:
loop should be used for repetition. the code below worked.
```
---

- name: "Configure {{ item.file }}"
  template:
    src: "{{ item.name }}-site.xml.j2"
    dest: "{{ item.file }}"
    validate: "xmllint %s"
  when:
    - item.file is defined
    - item.file | length > 0
  register: hadoop_site
  notify:
    - Restart namenode
  with_items:
    - { name: 'core', file: "{{ hadoop_core_site_file }}" }
    - { name: 'hdfs', file: "{{ hadoop_hdfs_site_file }}" }
    - { name: 'mapred', file: "{{ hadoop_mapred_site_file }}" }
    - { name: 'yarn', file: "{{ hadoop_yarn_site_file }}" }

# XXX restart hadoop services when hadoop_config_changed is true
# you cannot "notify" here because zkfc is not enabled when standalone
- set_fact:
    hadoop_config_changed: true
  when: hadoop_site.changed or (hadoop_config_changed is defined and hadoop_config_changed)
```